fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""Utility class for plotting metrics with error ranges."""
+
+from typing import List, Union
+import pandas as pd
+import numpy as np
+
+from ._metric_frame import MetricFrame
+from matplotlib.lines import Line2D
+from matplotlib.axes._axes import Axes
+
+_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR = \
+    "Ambiguous Error Metric. Please only provide one of: error_bars or conf_intervals."
+_METRIC_AND_ERROR_BARS_NOT_SAME_LENGTH_ERROR = \
+    "The list of metrics and list of error_bars are not the same length."
+_METRIC_LENGTH_ZERO_ERROR = \
+    "No metrics were provided to plot. A nonzero number of metrics is required."
+_METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR = \
+    "The list of metrics and list of conf_intervals are not the same length."
+
+_METRICS_NOT_LIST_OR_STR_ERROR = \
+    """Metric should be a string of a single metric or a list of the metrics
+     in provided MetricFrame, but {0} was provided"""
+_ERROR_BARS_MUST_BE_TUPLE = "Calculated error_bars must be tuple of length 2"
+_CONF_INTERVALS_MUST_BE_TUPLE = "Calculated conf_intervals must be tuple of length 2"
+_ERROR_BARS_NEGATIVE_VALUE_ERROR = "Calculated relative errors from error_bars must be positive"
+_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR = \
+    "Calculated conf_intervals' upper bound cannot be less than lower bound"
+
+
+def _check_if_metrics_and_error_metrics_same_length(metrics, error_bars, conf_intervals):
+    if error_bars is not None:
+        if len(error_bars) != len(metrics):
+            raise ValueError(_METRIC_AND_ERROR_BARS_NOT_SAME_LENGTH_ERROR)
+    elif conf_intervals is not None:
+        if len(conf_intervals) != len(metrics):
+            raise ValueError(_METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR)
+
+
+def _check_if_metrics_length_zero(metrics):
+    if len(metrics) == 0:
+        raise ValueError(_METRIC_LENGTH_ZERO_ERROR)
+
+
+def _check_for_ambiguous_error_metric(error_bars, conf_intervals):
+    # ambiguous error metric if both are provided
+    if error_bars is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+
+
+def _check_for_valid_metrics_format(metrics):
+    # ensure metrics is either list, str, or None
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+
+
+def _check_valid_error_bars(df_error_bars):
+    for tup in df_error_bars:
```

Comment:
I'd much prefer if errors were scalars (interpreted as symmetric errors). Possibly, we could also allow providing asymmetric errors as arrays or tuples of size two, but I'm not so eager to do that (see my comment below).